GPU/TPCFastTransformation/macro/TPCFastTransformInit.C

```diff
@@ -75,30 +82,107 @@ void TPCFastTransformInit(const char* fileName = "debugVoxRes.root",
+
+    for (auto y2x : y2xBins) {
+      std::cout << " y2x: " << y2x << std::endl;
+    }
+    std::cout << std::endl;
+
+    int nY2Xbins = trackResiduals.getNY2XBins();
+
+    std::cout << " TrackResiduals y2x bins: " << nY2Xbins << std::endl;
+    for (int i = 0; i < nY2Xbins; i++) {
+      std::cout << "scaled getY2X(bin) : " << trackResiduals.getY2X(0, i) / trackResiduals.getMaxY2X(0) << std::endl;
+    }
+
+    std::cout << "voxel tree z2xBins: " << z2xBins.size() << std::endl;
+
+    for (auto z2x : z2xBins) {
+      std::cout << "z2x: " << z2x << std::endl;
+    }
+    std::cout << std::endl;
+
+    int nZ2Xbins = trackResiduals.getNZ2XBins();
+    std::cout << " TrackResiduals z2x bins: " << nZ2Xbins << std::endl;
+    for (int i = 0; i < nZ2Xbins; i++) {
+      std::cout << "getZ2X(bin) : " << trackResiduals.getZ2X(i) << std::endl;
+    }
+  }
 
   std::cout << "create fast transformation ... " << std::endl;
 
   auto* helper = o2::tpc::TPCFastTransformHelperO2::instance();
 
   o2::tpc::TPCFastSpaceChargeCorrectionHelper* corrHelper = o2::tpc::TPCFastSpaceChargeCorrectionHelper::instance();
 
+  corrHelper->setNthreadsToMaximum();
+
   auto corrPtr = corrHelper->createFromTrackResiduals(trackResiduals, voxResTree, useSmoothed, invertSigns);
 
   std::unique_ptr<o2::gpu::TPCFastTransform> fastTransform(
     helper->create(0, *corrPtr));
 
-  o2::gpu::TPCFastSpaceChargeCorrection& corr = fastTransform->getCorrection();
-
   std::cout << "... create fast transformation completed " << std::endl;
 
   if (*outFileName) {
     fastTransform->writeToFile(outFileName, "ccdb_object");
   }
 
+  if (1) { // read transformation from the file
+
+    // const char* fileName = "master/out.root";
+
+    const char* fileName = outFileName;
+
+    std::cout << "load corrections from file " << fileName << std::endl;
+
+    fastTransform->cloneFromObject(*TPCFastTransform::loadFromFile(fileName, "ccdb_object"), nullptr);
+
+    o2::gpu::TPCFastSpaceChargeCorrection& corr = fastTransform->getCorrection();
+
+    if (0) {
+      std::cout << "check the loaded correction ..." << std::endl;
+
+      const o2::gpu::TPCFastTransformGeo& geo = helper->getGeometry();
+
+      // for (int iRoc = 0; iRoc < geo.getNumberOfSlices(); iRoc++) {
+      for (int iRoc = 0; iRoc < 1; iRoc++) {
+        for (int iRow = 0; iRow < geo.getNumberOfRows(); iRow++) {
+          auto& info = corr.getSliceRowInfo(iRoc, iRow);
+          std::cout << "roc " << iRoc << " row " << iRow
+                    << " gridV0 " << info.gridV0 << " gridCorrU0 " << info.gridCorrU0 << " gridCorrV0 " << info.gridCorrV0
+                    << " scaleCorrUtoGrid " << info.scaleCorrUtoGrid << " scaleCorrVtoGrid " << info.scaleCorrVtoGrid
+                    << " gridU0 " << info.gridU0 << " scaleUtoGrid " << info.scaleUtoGrid << " scaleVtoGrid " << info.scaleVtoGrid
+                    << std::endl;
+        }
+      }
+    }
+  }
+
```

Comment:
Why are the data written to file and read back?
Is this only for debugging? If so, please add a comment.

Reply:
@sgorbuno , can you comment on this?